File: tests/ruby/client_test.rb
# Ruby Client Tests
# Run: ruby tests/ruby/client_test.rb

require 'minitest/autorun'
require_relative '../../generated/ruby/lib/solana_rpc_client'

class ClientTest < Minitest::Test
  TEST_PUBKEY = 'vines1vzrYbzLMRdu58ou5XTby4qAqVRLmqo36NKPTg'

  def setup
    @client = Solana::RpcClient.new(Solana::RpcClient::DEVNET)
  end

  def test_get_version
    result = @client.get_version
    assert result.key?('solana-core')
    assert_kind_of String, result['solana-core']
  end

  def test_get_health
    result = @client.get_health
    assert_includes %w[ok behind unknown], result
  end

  def test_get_slot
    result = @client.get_slot
    assert_kind_of Integer, result
    assert result > 0
  end

  def test_get_block_height
    result = @client.get_block_height
    assert_kind_of Integer, result
    assert result > 0
  end

  def test_get_epoch_info
    result = @client.get_epoch_info
    assert result.key?('epoch')
    assert result.key?('slotIndex')
    assert result['epoch'] > 0
  end

  def test_get_balance
    result = @client.get_balance(TEST_PUBKEY)
    assert result.key?('context')
    assert result.key?('value')
    assert_kind_of Integer, result['value']
  end

  def test_get_account_info
    result = @client.get_account_info(TEST_PUBKEY)
    assert result.key?('value')
    assert result['value'].key?('lamports')
    assert result['value'].key?('owner')
  end

  def test_get_latest_blockhash
    result = @client.get_latest_blockhash
    assert result.key?('value')
    assert result['value'].key?('blockhash')
  end

  def test_get_genesis_hash
    result = @client.get_genesis_hash
    assert_kind_of String, result
    assert result.length > 30
  end

  def test_get_supply
    result = @client.get_supply
    assert result.key?('value')
    assert result['value'].key?('total')
  end

  def test_invalid_pubkey
    assert_raises(RuntimeError) do
      @client.get_balance('invalid')
    end
  end
end

File: tests/php/ClientTest.php
<?php
/**
 * PHP Client Tests
 * Run: phpunit tests/php/ClientTest.php
 */

require_once __DIR__ . '/../../generated/php/src/SolanaRpcClient.php';

use PHPUnit\Framework\TestCase;
use Solana\Rpc\SolanaRpcClient;

class ClientTest extends TestCase
{
    private SolanaRpcClient $client;
    private string $testPubkey = 'vines1vzrYbzLMRdu58ou5XTby4qAqVRLmqo36NKPTg';

    protected function setUp(): void
    {
        $this->client = new SolanaRpcClient(SolanaRpcClient::DEVNET);
    }

    public function testGetVersion(): void
    {
        $result = $this->client->getVersion();
        $this->assertArrayHasKey('solana-core', $result);
        $this->assertIsString($result['solana-core']);
    }

    public function testGetHealth(): void
    {
        $result = $this->client->getHealth();
        $this->assertContains($result, ['ok', 'behind', 'unknown']);
    }

    public function testGetSlot(): void
    {
        $result = $this->client->getSlot();
        $this->assertIsInt($result);
        $this->assertGreaterThan(0, $result);
    }

    public function testGetBlockHeight(): void
    {
        $result = $this->client->getBlockHeight();
        $this->assertIsInt($result);
        $this->assertGreaterThan(0, $result);
    }

    public function testGetEpochInfo(): void
    {
        $result = $this->client->getEpochInfo();
        $this->assertArrayHasKey('epoch', $result);
        $this->assertArrayHasKey('slotIndex', $result);
        $this->assertGreaterThan(0, $result['epoch']);
    }

    public function testGetBalance(): void
    {
        $result = $this->client->getBalance($this->testPubkey);
        $this->assertArrayHasKey('context', $result);
        $this->assertArrayHasKey('value', $result);
        $this->assertIsInt($result['value']);
    }

    public function testGetAccountInfo(): void
    {
        $result = $this->client->getAccountInfo($this->testPubkey);
        $this->assertArrayHasKey('value', $result);
        $this->assertArrayHasKey('lamports', $result['value']);
        $this->assertArrayHasKey('owner', $result['value']);
    }

    public function testGetLatestBlockhash(): void
    {
        $result = $this->client->getLatestBlockhash();
        $this->assertArrayHasKey('value', $result);
        $this->assertArrayHasKey('blockhash', $result['value']);
    }

    public function testGetGenesisHash(): void
    {
        $result = $this->client->getGenesisHash();
        $this->assertIsString($result);
        $this->assertGreaterThan(30, strlen($result));
    }

    public function testGetSupply(): void
    {
        $result = $this->client->getSupply();
        $this->assertArrayHasKey('value', $result);
        $this->assertArrayHasKey('total', $result['value']);
    }

    public function testInvalidPubkey(): void
    {
        $this->expectException(\Exception::class);
        $this->client->getBalance('invalid');
    }
}
